src/components/Form/SelectFIeld.tsx

```diff
@@ -0,0 +1,40 @@
+import './formFields.css';
+import Options from '../../helper/selectOptions';
+
+type SelectProps<T> = {
+  label: string;
+  value: T;
+  name: string;
+  placeholder: string;
+  error: boolean;
+  options: T[];
+  onChange: (e: React.ChangeEvent<HTMLSelectElement>) => void;
+};
+
+const SelectField = <T extends string>(props: SelectProps<T>): JSX.Element => {
```

Comment:
I would preffer not to define T as extend string as it could be any object. So you can have comma back it is fine. You need maybe here than probably another prop: getOptionValue: (option: T) => string.

This you pass to select to extract option value as string. So in case you give select array of objects, you also give this callback for extracting value from it as string.  And maybe in additional getOptionName: (option: T) => string. 

So you can extract two values one key value for identifier and one for display in select. 

Let say you want to show select which selects Certificates for example. so you have 

interace Certificate {
id: number; 
name: string;
type: CertificateType;
...
}

And you now want select of Certificates:

const someCertificates: Certificate[] = []; // assume it's not empty array

<Select<Cetificate>
   options={someCertificates}
   getOptionValue={(option: Certificate) => option.id} // ignore using lambda directly in JSX :) 
   getOptionName={(option: Certificate) => option.name})
   ....
/>

Than in select rendering option would be:

<option value={props.getOptionValue(option)}>{getOptionName(option)}</option>

in this way your Select is completelly generic